src/runtime/virtcontainers/sandbox.go

```diff
@@ -688,6 +689,9 @@ func (s *Sandbox) createResourceController() error {
 		//TODO: in Docker or Podman use case, it is reasonable to set a constraint. Need to add a flag
 		// to allow users to configure Kata to constrain CPUs and Memory in this alternative
 		// scenario. See https://github.com/kata-containers/runtime/issues/2811
+	} else {
+		// Generate a random string as the cgroup path to prevent duplicates.
+		cgroupPath = uuid.Generate().String()
```

Comment:
Perhaps a more correct way is to report an error?

Reply:
How did you reproduce it? I wonder if it is because some container annotation is not set properly. We should always be able to find the podsandbox container when creating a new sandbox in the current kata workflow.

Reply:
I have the same issue.  
just use `ctr run --runtime "io.containerd.kata.v2"`, the `containerType` will be `singleContainer`, so `cgroupPath` will be `""`. 

why choose to use a generated random string instead of sandbox.id? 
As far as I know: 

- using randStr, the target cgroup is '/sys/fs/cgroup/vc/{randStr}'; 
- using `sandbox.Id`, the target cgroup is '/sys/fs/cgroup/vc/{sandbox.Id}';

Reply:
@bergwolf Yes, it should be that SingleContainer was missed when judging the ContainerType of the annotation.

Reply:
@smile-luobin Yes, it is better to directly use the sandbox id as the cgroup path, I modified it.